include/boost/geometry/algorithms/detail/disjoint/multipoint_geometry.hpp

```diff
@@ -93,30 +94,27 @@ class multipoint_multipoint
                              MultiPoint2 const& multipoint2,
                              Strategy const&)
     {
-        typedef typename Strategy::cs_tag cs_tag;
-        typedef geometry::less<void, -1, cs_tag> less_type;
-
         BOOST_GEOMETRY_ASSERT( boost::size(multipoint1) <= boost::size(multipoint2) );
 
-        typedef typename boost::range_value<MultiPoint1>::type point1_type;
+        using cs_tag = typename Strategy::cs_tag;
+        using less_type = geometry::less<void, -1, cs_tag>;
+        using point1_type = typename boost::range_value<MultiPoint1>::type;
 
         std::vector<point1_type> points1(boost::begin(multipoint1),
                                          boost::end(multipoint1));
 
         std::sort(points1.begin(), points1.end(), less_type());
 
-        typedef unary_disjoint_predicate
+        using predicate_type = unary_disjoint_predicate
             <
                 typename std::vector<point1_type>::const_iterator,
                 cs_tag
-            > predicate_type;
+            >;
 
-        return check_iterator_range
-            <
-                predicate_type
-            >::apply(boost::begin(multipoint2),
-                     boost::end(multipoint2),
-                     predicate_type(points1.begin(), points1.end()));
+        return boost::end(multipoint2) == find_if(
+            boost::begin(multipoint2),
+            boost::end(multipoint2),
+            predicate_type(points1.begin(), points1.end()));
```

Comment:
Here we search for intersecting points and if they are not found then the result of disjoint is `true`. If multipoint is empty they are disjoint so ok.